FEXCore/Source/Interface/Core/JIT/JIT.cpp

```diff
@@ -484,11 +484,12 @@ static void IndirectBlockDelinker(FEXCore::Core::CpuStateFrame* Frame, FEXCore::
 
 static uint64_t Arm64JITCore_ExitFunctionLink(FEXCore::Core::CpuStateFrame* Frame, FEXCore::Context::ExitFunctionLinkData* Record) {
   auto Thread = Frame->Thread;
+  bool TFSet = Thread->CurrentFrame->State.flags[X86State::RFLAG_TF_RAW_LOC];
+  uintptr_t HostCode {};
   auto GuestRip = Record->GuestRIP;
 
-  auto HostCode = Thread->LookupCache->FindBlock(GuestRip);
-
-  if (!HostCode) {
+  if (TFSet || !(HostCode = Thread->LookupCache->FindBlock(GuestRip))) {
```

Comment:
I get that this is convenient when writing code, but for the sake of readability let's not combine control flow with short-circuiting behavior and variable initializations.

Reply:
Fair enough :)